javalanglib/src/main/scala/java/lang/_String.scala

```diff
@@ -734,6 +734,212 @@ for (cp <- 0 to Character.MAX_CODE_POINT) {
     }
   }
 
+  def stripLeading(): String = {
+    length() match {
+      case 0                                                => thisString
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var idx = 0
+        while (idx < n && Character.isWhitespace(charAt(idx))) idx += 1
+        substring(idx)
+    }
+  }
+
+  def stripTrailing(): String = {
+    length() match {
+      case 0                                                => thisString
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var idx = n - 1
+        while (idx >= 0 && Character.isWhitespace(charAt(idx))) idx -= 1
+        substring(0, idx + 1)
+    }
+  }
+
+  def strip(): String = {
+    length() match {
+      case 0                                                => thisString
+      case 1 if java.lang.Character.isWhitespace(charAt(0)) => ""
+      case n                                                =>
+        var leading = 0
+        while (leading < n && Character.isWhitespace(charAt(leading))) leading += 1
+        if (leading == n) ""
+        else {
+          var trailing = n - 1
+          while (trailing >= 0 && Character.isWhitespace(charAt(trailing))) trailing -= 1
+          if (leading == 0 && trailing + 1 == n) thisString
+          else substring(leading, trailing + 1)
+        }
+    }
+  }
+
+  def isBlank(): scala.Boolean = {
+    val len = length()
+    var start = 0
+    while (start != len && java.lang.Character.isWhitespace(charAt(start)))
+      start += 1
+    start == len
+  }
+
+  private def splitLines(): js.Array[String] = {
+    val xs   = js.Array[String]()
+    val len  = length()
+    var idx  = 0
+    var last = 0
+    while (idx < len) {
+      val c = charAt(idx)
+      if (c == '\n' || c == '\r') {
+        xs.push(substring(last, idx))
+        if (c == '\r' && idx + 1 < len && charAt(idx + 1) == '\n') idx += 1
+        last = idx + 1
+      }
+      idx += 1
+    }
+    // make sure we add the last segment, but not the last new line
+    if(last  != len) xs.push(substring(last))
+    xs
+  }
+
+  def indent(n: Int): String = {
+    def forEachLn(f: js.Function1[String, String]): String = {
+      var out = ""
+      var i   = 0
+      val xs  = splitLines()
+      while (i < xs.length) {
+        out += f(xs(i)) + "\n"
+        i += 1
+      }
+      out
+    }
+    n match {
+      case remove if remove < 0 =>
+        forEachLn { l =>
+          var idx = 0
+          val lim = if (l.length > remove) l.length else remove
+          while (idx < lim && Character.isWhitespace(l.charAt(idx))) idx += 1
+          l.substring(idx)
+        }
+      case insert if insert > 0 =>
+        val padding = " ".repeat(insert)
```

Comment:
Thanks for the PR.

You'll probably want to check out our [coding style document](https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md). Alternatively, when everything else is OK, I can fix the coding style myself and force-push the branch.

For this line to compile on JDK 8, you will need to write it as
```suggestion
        val padding = " ".asInstanceOf[_String].repeat(insert)
```
There are also two calls to `isBlank()` that will require the same treatment.

Reply:
Sorry about the styling! I'm quite used to scalafmt so was useless at formatting by hand.
I'll fix the JDK 8 issues first and format it properly.